src/main/java/vazkii/quark/base/recipe/PotionIngredient.java

```diff
@@ -68,23 +69,27 @@ public JsonElement serialize() {
         @Nonnull
         @Override
         public PotionIngredient parse(@Nonnull PacketBuffer buffer) {
-            Item item = ForgeRegistries.ITEMS.getValue(buffer.readResourceLocation());
-            Potion potion = ForgeRegistries.POTION_TYPES.getValue(buffer.readResourceLocation());
+            ResourceLocation itemName = buffer.readResourceLocation();
+            Item item = Registry.ITEM.getOptional(itemName).orElseThrow(() -> new InvalidParameterException("Could not find item with registry name '" + itemName + "'"));
+            ResourceLocation potionName = buffer.readResourceLocation();
+            Potion potion = Registry.POTION.getOptional(potionName).orElseThrow(() -> new InvalidParameterException("Could not find potion with registry name '" + potionName + "'"));
```

Comment:
I'm fine with just using get() here for more concise code. The packets should always be well formed and if they aren't we have bigger problems.